server/consumer.go

```diff
@@ -1605,13 +1623,8 @@ func (o *consumer) sendDeleteAdvisoryLocked() {
 		Domain:   o.srv.getOpts().JetStreamDomain,
 	}
 
-	j, err := json.Marshal(e)
```

Comment:
I think I did this here on purpose..

Reply:
I'm not sure it matters where the marshaling takes place, but if we can avoid doing it until after the interest check, that saves us some CPU time & allocs.

Reply:
We should do it after the check for sure. But I need to recall why I put it there? Maybe use more cores?